# A service call to carry out an inverse kinematics computation with extended IK pose
# The inverse kinematics request
moveit_msgs/PositionIKRequest ik_request
# A set of constraints that the IK must obey
moveit_msgs/Constraints constraints
#extended pose of the end effector
geometry_msgs/Pose ik_pose
#obey constraints?
bool constraint_aware
# Maximum allowed time for IK calculation
duration timeout
---
# The returned solution 
moveit_msgs/RobotState solution
moveit_msgs/MoveItErrorCodes error_code
